From 586e222f0c4cd7cb400e6dbb3dd379ada550af87 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 21 Jun 2018 17:12:19 +0000
Subject: [PATCH] Temporary fix for SIGBUS crash: disable TLS

A conflict occurs between Qt threads and bionic threads; as a
temporary fix, disable TLS in Qt to avoid such conflict.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/corelib/thread/qthread_unix.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/thread/qthread_unix.cpp b/src/corelib/thread/qthread_unix.cpp
index 710528ea41..42b1885aae 100644
--- a/src/corelib/thread/qthread_unix.cpp
+++ b/src/corelib/thread/qthread_unix.cpp
@@ -110,7 +110,7 @@ enum { ThreadPriorityResetFlag = 0x80000000 };
 
 #if defined(Q_OS_LINUX) && defined(__GLIBC__) && (defined(Q_CC_GNU) || defined(Q_CC_INTEL)) && !defined(QT_LINUXBASE)
 /* LSB doesn't have __thread, https://lsbbugs.linuxfoundation.org/show_bug.cgi?id=993 */
-#define HAVE_TLS
+#undef HAVE_TLS
 #endif
 #if defined(Q_CC_XLC) || defined (Q_CC_SUN)
 #define HAVE_TLS
